src/Nox.ClientApi.Tests/Tests/Controllers/StoresControllerTests.cs

```diff
@@ -143,6 +144,142 @@ public async Task Get_StoreOwnerOdataQuery_ReturnOwner()
+                    Neighborhood: null!,
+                    AdministrativeArea1: null!,
+                    AdministrativeArea2: null!,
+                    PostalCode: "KT16 0RS",
+                    CountryId: CountryCode.GB),
+                Location = new LatLongDto(51.3728033, -0.5389749),
+                LicenseId = licensePostResponse!.Id
+            };
+
+            // Act
+            var result = await PostAsync<StoreCreateDto, StoreDto>(EntityUrl, storeCreateDto);
+
+            const string oDataRequest = $"$expand={nameof(StoreDto.License)}";
+            var response = await GetODataSimpleResponseAsync<StoreDto>($"{EntityUrl}/{result!.Id}?{oDataRequest}");
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Should()
+                .BeOfType<StoreDto>()
+                .Which.Id.Should().NotBeEmpty();
+            response.Should().NotBeNull();
+            response!.License.Should().NotBeNull();
+            response!.License!.Id.Should().Be(licensePostResponse!.Id);
+            response!.License!.Issuer.Should().Be(licenseName);
+        }
+        #endregion
+
+        #region POST Entity with Invalid RelationshipId /api/{EntityPluralName} => api/stores
+        [Fact]
+        public async Task Post_WithInvalidRelationshipId_CreatesEntityWithoutRelationship()
+        {
+            // Arrange            
+            var storeCreateDto = new StoreCreateDto
+            {
+                Name = _fixture.Create<string>(),
+                Address = new StreetAddressDto(
+                    StreetNumber: null!,
+                    AddressLine1: "3000 Hillswood Business Park",
+                    AddressLine2: null!,
+                    Route: null!,
+                    Locality: null!,
+                    Neighborhood: null!,
+                    AdministrativeArea1: null!,
+                    AdministrativeArea2: null!,
+                    PostalCode: "KT16 0RS",
+                    CountryId: CountryCode.GB),
+                Location = new LatLongDto(51.3728033, -0.5389749),
+                LicenseId = _fixture.Create<int>()
+            };
+
+            // Act
+            var result = await PostAsync<StoreCreateDto, StoreDto>(EntityUrl, storeCreateDto);
+
+            const string oDataRequest = $"$expand={nameof(StoreDto.License)}";
+            var response = await GetODataSimpleResponseAsync<StoreDto>($"{EntityUrl}/{result!.Id}?{oDataRequest}");
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Should()
+                .BeOfType<StoreDto>()
+                .Which.Id.Should().NotBeEmpty();
+            response.Should().NotBeNull();
+            response!.License.Should().BeNull();
+        }
+        #endregion
+
+        #region POST Entity with Deleted RelationshipId /api/{EntityPluralName} => api/stores
+        [Fact]
+        public async Task Post_WithDeletedRelationshipId_ShouldNotCreateRefToRelatedEntity()
```

Comment:
@rochar 
I've just realized that we don't check in CreateRef endpoints. So if the user tries to create a ref between the existing and the deleted entity, the reference will be created. Should we have a ticket to fix it?

Also I'm not sure what happens if we get the entity, expand relationships and some of them are removed (do we filter them out?).

Reply:
That function receives an entity, so that entity is new or already exists, so we should not need to verify if is delete, the caller should not set a deleted entity, this should not be possible since delete only exists in a database level for auditing

I have created this issue #890 so we do not need to care about this when querying entities or dtos. This will also take care of expanding the relations